src/index.ts

```diff
@@ -185,12 +185,39 @@ function createMultipartStream(boundary: string, multipart: RequestPart[]) {
   return stream;
 }
 
+function teenyRequest(url: string): Request;
+function teenyRequest(url: string, reqOpts: Options): Request;
+function teenyRequest(url: string, callback: RequestCallback): void;
+function teenyRequest(
+  url: string,
+  reqOpts: Options,
+  callback: RequestCallback
+): void;
 function teenyRequest(reqOpts: Options): Request;
 function teenyRequest(reqOpts: Options, callback: RequestCallback): void;
 function teenyRequest(
-  reqOpts: Options,
-  callback?: RequestCallback
+  urlOrOpts: Options | string,
+  optsOrCallback?: Options | RequestCallback,
+  cb?: RequestCallback
 ): Request | void {
+  let url: string | undefined = undefined;
+  let reqOpts: Options | undefined = undefined;
+  if (typeof urlOrOpts === 'string') {
+    url = urlOrOpts;
+  }
+  if (typeof urlOrOpts === 'object') {
```

Comment:
I wish there was a standard way to do this with less ritual 😆 one of my first attempts at open-source was doing just this, and I made a DSL that was even worse:

```js
exports.foo = function(path, options, fn){
   args([this, 'string1', ['object1', 'function1'], 'function1'], function() {
   });
};
```

yargs has a library that asserts against parameters that's maybe a little closer to useful:

```js
  self.nargs = function (key, value) {
    argsert('<string|object|array> [number]', [key, value], arguments.length)
    populateParserHintObject(self.nargs, false, 'narg', key, value)
    return self
  }
```

Any ways, this pattern is used a lot by folks, I wish we as a community could come up with a better approach (not a critique of this PR).